### PR TITLE
Refactor actions_list for clarity

### PR DESCRIPTION
**Pull request by mfurtak**
Addresses #4335

Removes the need for rubocop violation suppressions by refactoring `show_details` to use some helper methods.

Fixes a problem with trying to show the details of an action that is not yet properly subclassing `Action`, instead showing the helpful error given during action listing

screen shot 2016-04-23 at 2 24 35 pm

### CODE REVIEW

File: fastlane/lib/fastlane/documentation/actions_list.rb [hunk 1]
```diff
@@ -24,18 +22,18 @@ def self.print_all(platform: nil)
           current << authors.first.green if authors.count == 1
           current << "Multiple".green if authors.count > 1
         else
-          UI.error("Please update your action file #{name} to be a subclass of `Action` by adding ` < Action` after your class name.")
+          UI.error action_subclass_error(name)
           current << "Please update action file".red
+          current << ' '
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: Why is this necessary?

File: fastlane/lib/fastlane/documentation/actions_list.rb [hunk 2]
```diff
@@ -44,100 +42,119 @@ def self.print_all(platform: nil)
...
 
-        rows = []
-        rows << [action.description] if action.description
-        rows << [' ']
-        if action.details
-          rows << [action.details]
-          rows << [' ']
+      if action
+        unless action < Action
+          UI.user_error! action_subclass_error(filter)
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: Should this really be `user_error!` so it "crashes"? On line 25 we just show the error message

File: fastlane/spec/actions_list_spec.rb [hunk 3]
```diff
@@ -71,5 +71,17 @@
         end
       end
     end
+
+    describe "with a class in the Actions namespace that does not extend action" do
+      it "trying to show its details presents a helpful error message" do
+        require 'fixtures/broken_actions/broken_action.rb'
+
+        expect(UI).to receive(:user_error!).with(/be a subclass/).and_raise("boom")
```
Comment (KrauseFx) [inline fastlane/spec/actions_list_spec.rb]: 💥

**Comment by KrauseFx** [comment]
Looks great, thanks for cleaning up this class! I see in the PR description you explain the change with the not properly subclassing warning. The reason why it might not be a good idea to crash is that it's a breaking change: back in the days fastlane's action didn't have a standardized format. Someone might (not sure if true) still use an action that's using the old format, and hasn't yet had the chance to migrate.
However the warning has been shown for about a year now, so now might be a good time to not support them any more. It's just the documentation, so it doesn't actually break the production system.

File: fastlane/lib/fastlane/documentation/actions_list.rb (hunk 1 above)
Comment (mfurtak) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: Ah, the header references three columns, so I thought it made sense that the row should always have three entries.

File: fastlane/lib/fastlane/documentation/actions_list.rb (hunk 2 above)
Comment (mfurtak) [inline fastlane/lib/fastlane/documentation/actions_list.rb]: Given that there is improperly defined action:

When the user asks to see all of the actions, I think it makes sense that we display an error, but then also show all the other information that we can in the table. This was true before, and is still true with this change.

When the user is asking to see the details about the improperly defined action, the previous behavior crashed with a stack trace. These changes make it so that we display a helpful message and "crash" (so not a change in behavior with regard to success/failure).

I think it is helpful to stop executing (rather than showing the whole list) in this case, because the user asked for something specific, and we know what they're referring to, but it is broken and can't be displayed.

**Comment by mfurtak** [comment]
@KrauseFx If you would be in favor of removing the checks/errors about improperly defined actions, that's OK too. Removing them would make the logic even simpler. Let me know!

**Comment by mfurtak** [comment]
Discussed with @KrauseFx in Slack, where he approved!
